src/Model/ModelManagerInterface.php

```diff
@@ -97,135 +80,100 @@ public function createQuery($class, $alias = 'o');
      * composed of multiple columns. If you need a string representation,
      * use getNormalizedIdentifier resp. getUrlSafeIdentifier
      *
-     * @param object $model
-     *
-     * @return array list of all identifiers of this model
+     * @return mixed[]
      */
-    public function getIdentifierValues($model);
+    public function getIdentifierValues(object $model): array;
 
     /**
-     * Get a list of the field names models of the specified class use to store
-     * the identifier.
+     * Get a list of the field names models of the specified fully qualified
+     * class name used to store the identifier.
      *
-     * @param string $class fully qualified class name
-     *
-     * @return array
+     * @return string[]
      */
-    public function getIdentifierFieldNames($class);
+    public function getIdentifierFieldNames(?string $class): array;
```

Comment:
```suggestion
    public function getIdentifierFieldNames(string $class): array;
```

Reply:
If I change this one I have to change this other line too: 

https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Form/ChoiceList/ModelChoiceLoader.php#L83

This class is construct with $class and it tells on the phpdoc is a string, but it can accept null too, because of this: 

https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Form/Type/ModelType.php#L94-L100

Test break if I make you change here:

https://github.com/sonata-project/SonataAdminBundle/blob/3.x/tests/Form/Type/ModelTypeTest.php#L34-L88

It does not pass the class.

Question is: is that class a mandatory parameter there? Should I change the ModelChoiceLoader to initialize the identifier property only if the $class parameter is not null? This last one is maybe my best choice here, but not really sure. I agree on the change you propose, but not sure what is the best way to apply it.

thoughts?

Reply:
@jordisala1991 
Where is used the identifier property of the ModelChoiceLoader class ?

I find no usage.

Reply:
Find no usages to, it is a public property and I found it was added like 7 years ago, maybe on the persistence bundles?

Reply:
Found nothing in the storage bundle. Maybe we could remove this property. 

It's a way to solve your problem 😅

The public property was added 2 year after the definition in the construct.
I feel like it is something that was never used since the first day.

Reply:
It was not even used on the commit adding it: https://github.com/sonata-project/SonataAdminBundle/commit/3e7ad359411a5f4ae119d9dc3ef26b3e44716fd1

I will remove it and see what happens

Reply:
It was used here: https://github.com/sonata-project/SonataAdminBundle/commit/a92abd893a7d4ce727825d69b8152b24abfea659#diff-93cd6b7ac60ab3cbbb73a957c2f2f7a3R80

Maybe a bad copy paste.

By the way, `getIdentifierFieldNames` wont be use. I don't know if we should keep this in the interface.

`getModelIdentifier` is deprecated in favor of `getIdentifierFieldNames`, but the only usage I found was https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/3.x/src/Model/ModelManager.php#L607

Reply:
Done